src/meta/meta_bulk_load_service.cpp

```diff
@@ -283,41 +283,92 @@ void bulk_load_service::create_partition_bulk_load_dir(const std::string &app_na
 }
 
 // ThreadPool: THREAD_POOL_META_STATE
-void bulk_load_service::partition_bulk_load(const std::string &app_name, const gpid &pid)
+bool bulk_load_service::check_partition_status(
+    const std::string &app_name,
+    const gpid &pid,
+    bool always_unhealthy_check,
+    const std::function<void(const std::string &, const gpid &)> &retry_function,
+    /*out*/ partition_configuration &pconfig)
 {
-    FAIL_POINT_INJECT_F("meta_bulk_load_partition_bulk_load", [](dsn::string_view) {});
-
     std::shared_ptr<app_state> app = get_app(pid.get_app_id());
     if (app == nullptr || app->status != app_status::AS_AVAILABLE) {
         dwarn_f(
             "app(name={}, id={}) is not existed, set bulk load failed", app_name, pid.get_app_id());
         handle_app_unavailable(pid.get_app_id(), app_name);
-        return;
+        return false;
     }
 
-    rpc_address primary_addr = app->partitions[pid.get_partition_index()].primary;
-    if (primary_addr.is_invalid()) {
+    pconfig = app->partitions[pid.get_partition_index()];
+    if (pconfig.primary.is_invalid()) {
         dwarn_f("app({}) partition({}) primary is invalid, try it later", app_name, pid);
         tasking::enqueue(LPC_META_STATE_NORMAL,
                          _meta_svc->tracker(),
-                         std::bind(&bulk_load_service::partition_bulk_load, this, app_name, pid),
+                         [this, retry_function, app_name, pid]() { retry_function(app_name, pid); },
```

Comment:
why capture `this`

Reply:
Well, it is not necessary in this final version. How about delete it in my next pull request?

Reply:
ok